test/worldwide/address_test.rb

```diff
@@ -143,9 +143,110 @@ class AddressTest < ActiveSupport::TestCase
     end
 
     test "concatenated_address2 returns address2 concatenated with neighborhood separated by delimiter and decorator" do
-      address = Address.new(address2: "dpto 4", neighborhood: "Centro", country_code: "BR")
+      br_address = Address.new(address2: "dpto 4", neighborhood: "Centro", country_code: "BR")
+      ph_address = Address.new(address2: "apt 4", neighborhood: "294", country_code: "PH")
+      vn_address = Address.new(address2: "apt 4", neighborhood: "Cầu Giấy", country_code: "VN")
 
-      assert_equal "dpto 4, Centro", address.concatenated_address2
+      assert_equal "dpto 4, Centro", br_address.concatenated_address2
+      assert_equal "apt 4 Barangay 294", ph_address.concatenated_address2
+      assert_equal "apt 4, Quận Cầu Giấy", vn_address.concatenated_address2
+    end
+
+    test "parse_address1 returns empty hash when additional address fields are not defined for the country" do
+      address = Address.new(address1: "123 Main Street", country_code: "US")
+      expected_hash = {}
```

Comment:
Is an empty hash desirable? Or do we want to provide an address1 key/value?

Reply:
Thinking that since this address is valid, we should probably return `{ "address1": "..." }` even though there's no actual parsing happening. Based on the above tests for `concatenated_address1` we still return a string falling back to the existing raw address string, if I understood those tests correctly. Makes sense to me that we return valid data back if we have it.

Seems like a safe assumption to say "no additional address fields" implies "use an assumed default structure of one single `address1` field" in this case.

Reply:
I have a preference for returning an empty response, perhaps even a nil.

Assume we have this address object
```ruby
address = Address.new(address1: "123 Cannot Be Split", ...)
```

### returning a hash with `address1:` as the only key
```
split_address1 = address.split_address1 # => { address1: "123 Cannot Be Split" }

if split_address1.key?(:address1)
  # handle as standard address
else
  # handle as extended address
end
```

### returning `nil`
```
split_address1 = address.split_address1 # => nil

if split_address1
  # handle as extended address
else
  # handle as standard address
end
```
The second one is less surprising to my developer brain. The first form requires more work to evaluate and more knowledge of implementation details.